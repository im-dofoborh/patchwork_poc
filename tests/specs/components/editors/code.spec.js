import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueCodemirror from 'vue-codemirror';
import CodeEditor from '~/components/editors/Code';

describe('CodeEditor component', () => {
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('code-editor', CodeEditor);
    localVue.use(VueCodemirror);
    wrapper = shallowMount(CodeEditor, {
      localVue,
    });
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders disabled variant', () => {
    wrapper.setProps({
      disabled: true,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
