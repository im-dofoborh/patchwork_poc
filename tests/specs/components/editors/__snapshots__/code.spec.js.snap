// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeEditor component renders default variant 1`] = `
<div class="sg-code-editor">
  <div class="sg-code-editor__editable">
    <div class="sg-code-editor__editable-text">
      Editable
    </div>
    <div class="sg-code-editor__editable-dot" style="background: green;"></div>
  </div>
  <codemirror-stub value="" name="codemirror" placeholder="" options="[object Object]" events="" globaloptions="[object Object]" globalevents=""></codemirror-stub>
</div>
`;

exports[`CodeEditor component renders disabled variant 1`] = `
<div class="sg-code-editor">
  <div class="sg-code-editor__editable">
    <div class="sg-code-editor__editable-text">
      Not Editable
    </div>
    <div class="sg-code-editor__editable-dot" style="background: orange;"></div>
  </div>
  <codemirror-stub value="" name="codemirror" placeholder="" options="[object Object]" events="" globaloptions="[object Object]" globalevents=""></codemirror-stub>
</div>
`;
